### PR TITLE
added command to generate simple scaffold for a component

### PR DESCRIPTION
**Pull request by naylor-b**
- current implementation just generates the component file and a test file.  It may be worthwhile to add the ability to create a plugin (setup.py file, etc.) in a later story.
- also updated the top level help for the 'openmdao' command.

### CODE REVIEW

File: openmdao/utils/scaffolding_templates/implicit_comp_template [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        """
+        Add inputs and outputs and declare partials.
+        """
+        # self.add_input(name='foo', shape=None, units=None)
+        # self.add_output(name='bar', shape=None, units=None)
+
+        # self.declare_partials(of='bar', wrt='foo', rows=None, cols=None, val=None)
+
+        # # Set up the derivatives.
+        # self.declare_partials(of='bar', wrt='foo', rows=None, cols=None, val=None)
```
Comment (swryan) [inline openmdao/utils/scaffolding_templates/implicit_comp_template]: looks like you have some unintended duplication here, as well as a commented comment

**Comment by swryan** [comment]
No documentation for this new command?  The existing `Command Line Tools` doc is under `Debugging` which doesn't quite seem appropriate for this one...

File: openmdao/utils/scaffolding_templates/implicit_comp_template (hunk 1 above)
Comment (naylor-b) [reply openmdao/utils/scaffolding_templates/implicit_comp_template]: fixed

**Comment by naylor-b** [comment]
added docs to the Command Line Tools section and moved that section under "Other Useful Docs".

**Pull request merged**